AmplifyPlugins/API/Sources/AWSAPIPlugin/Support/Utils/RESTOperationRequestUtils.swift

```diff
@@ -57,19 +57,31 @@ final class RESTOperationRequestUtils {
                                     headers: [String: String]?,
                                     requestPayload: Data?) -> URLRequest {
 
+        let baseHeaders = ["content-type": "application/json"]
         var baseRequest = URLRequest(url: url)
-        var requestHeaders = ["content-type": "application/json"]
-        if let headers = headers {
-            for (key, value) in headers {
-                requestHeaders[key] = value
-            }
-        }
-        baseRequest.allHTTPHeaderFields = requestHeaders
+        baseRequest = applyCustomizeRequestHeaders(
+            baseHeaders.merging(headers ?? [:], uniquingKeysWith: { _, new in new }),
+            on: baseRequest
+        )
```

Comment:
in this flow (construct -> intercept -> apply customized header), do we still need to apply the `request.headers` here?

Reply:
As [this method](https://github.com/aws-amplify/amplify-swift/blob/main/AmplifyPlugins/API/Sources/AWSAPIPlugin/Support/Utils/RESTOperationRequestUtils.swift#L55-L58) signature implicates, I think we still need to apply the headers.
```
static func constructURLRequest(with url: URL,
                                    operationType: RESTOperationType,
                                    headers: [String: String]?,
                                    requestPayload: Data?) -> URLRequest
```